lightning/src/ln/channelmanager.rs

```diff
@@ -8931,6 +9013,140 @@ where
+					match chan.splice_init(our_funding_contribution, &self.signer_provider, &self.entropy_source, self.get_our_node_id(), &self.logger) {
+						Ok(splice_ack_msg) => {
+							peer_state.pending_msg_events.push(events::MessageSendEvent::SendSpliceAck {
+								node_id: *counterparty_node_id,
+								msg: splice_ack_msg,
+							});
+						},
+						Err(err) => {
+							return Err(MsgHandleErrInternal::from_chan_no_close(err, msg.channel_id));
+						}
+					}
+				} else {
+					return Err(MsgHandleErrInternal::send_err_msg_no_close("Channel is not funded, cannot be spliced".to_owned(), msg.channel_id));
+				}
+			},
+		};
+
+		// TODO(splicing):
+		//  Change channel, change phase (remove and add)
+		//  Create new post-splice channel
+		//  etc.
+
+		Ok(())
+	}
+
+	/// Handle incoming splice request ack, transition channel to splice-pending (unless some check fails).
+	#[cfg(splicing)]
+	fn internal_splice_ack(&self, counterparty_node_id: &PublicKey, msg: &msgs::SpliceAck) -> Result<(), MsgHandleErrInternal> {
+		// TODO(splicing): checks
+		// TODO(splicing): check if we have initiated splicing
+
+		let per_peer_state = self.per_peer_state.read().unwrap();
+		let peer_state_mutex = per_peer_state.get(counterparty_node_id)
+			.ok_or_else(|| {
+				debug_assert!(false);
+				MsgHandleErrInternal::send_err_msg_no_close(format!("Can't find a peer matching the passed counterparty node_id {}", counterparty_node_id), msg.channel_id)
+			})?;
+		let mut peer_state_lock = peer_state_mutex.lock().unwrap();
+		let peer_state = &mut *peer_state_lock;
+
+		// Look for the channel
+		match peer_state.channel_by_id.entry(msg.channel_id) {
+			hash_map::Entry::Vacant(_) => return Err(MsgHandleErrInternal::send_err_msg_no_close(format!(
+					"Got a message for a channel from the wrong node! No such channel for the passed counterparty_node_id {}",
+					counterparty_node_id
+				), msg.channel_id)),
+			hash_map::Entry::Occupied(mut chan) => {
+				if let ChannelPhase::Funded(chan) = chan.get_mut() {
+					// check if splice is pending
+					let pending_splice = if let Some(pending_splice) = &chan.context.pending_splice_pre {
+						// Note: this is incomplete (their funding contribution is not set)
+						pending_splice.clone()
+					} else {
+						return Err(MsgHandleErrInternal::send_err_msg_no_close("Channel is not in pending splice".to_owned(), msg.channel_id));
+					};
+
+					let pre_channel_value = chan.context.get_value_satoshis();
+					let post_channel_value = PendingSpliceInfoPre::compute_post_value(pre_channel_value, pending_splice.our_funding_contribution, msg.funding_contribution_satoshis);
+
+					// Early check for reserve requirement, assuming maximum balance of full channel value
+					// This will also be checked later at tx_complete
+					let _res = chan.context.check_balance_meets_reserve_requirements(post_channel_value, post_channel_value)
+						.map_err(|e| MsgHandleErrInternal::from_chan_no_close(e, msg.channel_id))?;
```

Comment:
Similarly here? Seems there's a lot of back-and-forth with `channel` when it can be done in there instead.